test/TensorFlowRuntime/sese_loop_canonicalization.swift

```diff
@@ -45,4 +45,51 @@ ControlFlowTests.testAllBackends("natSumWithBreak") {
   expectEqualWithScalarTensor(5050, natSumWithBreak(200))
 }
 
+func sumOfProducts(_ M : Int32, _ N : Int32) -> Tensor<Float> {
+  // Effectively computes natSum(M)*natSum(N)
+  var sum = Tensor<Float>(0)
+  for i in 0..<M {
+    for j in 0..<N {
+      // i + 1 and j + 1 is a workaround for
+      // https://bugs.swift.org/browse/SR-8256
+      let prod = Tensor<Float>(Float((i + 1) * (j + 1)));
+      sum += prod
+    }
+  }
+  return sum
+}
+
+ControlFlowTests.testAllBackends("sumOfProducts") {
+  expectNearlyEqualWithScalarTensor(18.0, sumOfProducts(2, 3))
+  expectNearlyEqualWithScalarTensor(1980.0,sumOfProducts(10, 8))
+  expectNearlyEqualWithScalarTensor(11550.0,sumOfProducts(20, 10))
+}
+
+// A contrived example to test labeled breaks
+func sumOfProductsWithBound(_ M : Int32, _ N : Int32, _ Bound : Float) -> Tensor<Float> {
+  // Effectively computes natSum(M)*natSum(N) upto Bound
+  var sum = Tensor<Float>(0)
+outer_loop:for i in 0..<M {
+    for j in 0..<N {
+      // i + 1 and j + 1 is a workaround for
+      // https://bugs.swift.org/browse/SR-8256
+      let prod = Tensor<Float>(Float((i + 1) * (j + 1)));
+      sum += prod
+      if (sum.scalarized() >= Bound) {
```

Comment:
No need for semicolons or parens around conditions. This happens to me too when I write too much C++ :)